src/audio/types.h

```diff
@@ -244,6 +244,53 @@ class Bitrate {
 
 QDebug operator<<(QDebug dbg, Bitrate arg);
 
+// The BarLength is measured in beats and provides information
+// about the number expected within a single bar or phrase.
+class BarLength {
+  public:
+    typedef uint32_t value_t;
+
+    static BarLength valueFromUInt(unsigned int value) {
+        VERIFY_OR_DEBUG_ASSERT(value >= std::numeric_limits<value_t>::min() &&
+                value <= std::numeric_limits<value_t>::max()) {
+            return BarLength(kValueDefault);
```

Comment:
I think its better to return an invalid BarLength, if we were asked to construct it from an invalid value.
``suggestion
            return BarLength(0);
```